osquery/database/database.cpp

```diff
@@ -514,22 +515,58 @@ static Status migrateV0V1(void) {
   return Status();
 }
 
-Status upgradeDatabase() {
-  std::string db_results_version{""};
-  getDatabaseValue(kPersistentSettings, "results_version", db_results_version);
-
-  if (db_results_version == kDatabaseResultsVersion) {
-    return Status();
+Status upgradeDatabase(int to_version) {
+  std::string value;
+  Status st = getDatabaseValue(kPersistentSettings, kDbVersionKey, value);
```

Comment:
getDatabaseValue has two signatures, one of which is to get the int value.

Reply:
ok I see your problem  🥇

Reply:
Since the overload doesn't clearly define how to handle errors I prefer to do the conversion myself to distinguish between non-existing key, and key with invalid value. I could catch the exception thrown by `std::stoi` but depends on the implementation of the overload so I prefer not to.

Reply:
Yeah, that's what I thought. 👍